### PR TITLE
Show the limit in motors graph

### PR DESCRIPTION
**Pull request by McGiverGim**
Fixes https://github.com/betaflight/blackbox-log-viewer/issues/467

Is related to this firmware PR too https://github.com/betaflight/betaflight/pull/10106

I think this is what @ctzsnooze wants, but I'm not too sure because I don't have logs with motor limits enabled to see if it works in all cases.

I opened it as draft to be discussed to see if this is what we want to achieve, so ignore the code until discussed.

Everybody feel free to test and comment.

### CODE REVIEW

**Comment by ctzsnooze** [comment]
Thanks, McGiverGim!

The logs I have with motor_output_limit less than 100 appear to do the right thing at the higher end of the throttle.

At the low end, a log with dynamic and motor_output_limit shows a throttle setpoint of 6.2% (the configured idle percentage), but the motor percentage says 9% (probably from motor_output_limit) while the displayed motor percentage is shown as 0%.  

It would be good if the motors traces matched the percentage shown in the numbers.  

In other words, so that the displayed motor range is from 'absolute DShot zero' to 'absolute DShot max'.  In other words, motors should:

- be zero when disarmed, 
- rise to the idle value on arming, and then 
- move to the motor_output_limit at full throttle.

Setpoint should show the throttle position as a percentage, but the motors tab percentage should show absolute DShot percentage.

**Comment by McGiverGim** [comment]
@ctzsnooze I'm not sure if I understand you.

In my graphs I see the same value in the right panel, in the graph, or in the table for motors. And is the value that has been written into the blackbox.
I think maybe I don't understand what you are saying. Can you post some screen captures to see what is the problem? 

We will need too some logs with other different motor protocol (oneshot, multishot, proshot) to see what are the max and min values they use, I'm not too sure if they use the same scale than dshot. We don't need a real flight, a simple FC at the desk is enough, only some log with minimum and maximum throttle to see what they send. At this moment all my sdcard are damaged so I can't record them by myself.

**Comment by ctzsnooze** [comment]
I will make some better test log soon, apologies for being slow.

**Comment by McGiverGim** [comment]
How do you know that the motor graph shows a 0%? I think it is 8%. Can you add other line of something that goes from 0 to any other value to see if it goes under the motor line?

I have ignored the minimum value of 48 for motors, maybe the percentages are not accurate for this reason. I will change that and update the code. I suppose: 48 -> 0%, 2047 -> 100%.

About showing dshot value, I think is easier to users understand values from 0 to 100% and not from 48 to 2047. And more if it changes from an analog protocol to dshot. But this can be discussed later. It will not be difficult to show if needed.

**Comment by McGiverGim** [comment]
Ahh! I know, your setpoint is 6% and the motors 8%, and setpoint is over motor. Can you attach your log, please?

**Comment by ctzsnooze** [comment]
It is not an ideal log, because I can't remember if it it has motor output limit enabled...  It will do.  I will make another that has easier values to test with, from current master.  This is not from current master.
[7in.BBL.zip](https://github.com/betaflight/blackbox-log-viewer/files/5266237/7in.BBL.zip)

**Comment by McGiverGim** [comment]
@ctzsnooze fixed! Now it looks much better, please test the latest code. If it works ok, I think we have pending:
- A log with limit enabled, to see what the header looks like.
- Logs with other protocols, with and without limit.
- Discuss if we need to show a percentage or the dshot value directly.

**Comment by McGiverGim** [comment]
Fixed some offsets... another thing to decide if finally we stay with percentages: now is strange that the setpoint throttle uses one decimal, but the motors have no decimals. Maybe we need to unify both.

**Comment by McGiverGim** [comment]
Uploaded a new version that:
- Can work with analog and digital protocols
- Show next to the percentage the raw value (dshot or analog). In this way we can validate if my graphs are ok.

**Comment by McGiverGim** [comment]
After talking with @ctzsnooze it seems he is happy with the final result of this PR. So ready to be reviewed, modified if necessary and merged.
This PR is how @ctzsnooze and me see how the motors must be shown in the graphs, specially after this firmware PR: https://github.com/betaflight/betaflight/pull/10294

File: js/flightlog.js
Comment (haslinghuis) [inline js/flightlog.js:1159]: With a float we get more precision

PWM | Dshot | MotorPct
-- | -- | --
1000 | 0 | -2.40120
1001 | 48 | 0.00000
1002 | 50 | 0.10010
1003 | 52 | 0.20020
1004 | 54 | 0.30030
1005 | 56 | 0.40040
1006 | 58 | 0.50050
1007 | 60 | 0.60060
1008 | 62 | 0.70070
1009 | 64 | 0.80080
1010 | 66 | 0.90090
1011 | 68 | 1.00100
1012 | 70 | 1.10110
1013 | 72 | 1.20120
1014 | 74 | 1.30130
1015 | 76 | 1.40140

File: js/flightlog_fields_presenter.js
Comment (haslinghuis) [inline js/flightlog_fields_presenter.js:514]: With a float we get more precision?

PWM | Dshot | MotorPct
-- | -- | --
1000 | 0 | -2.40120
1001 | 48 | 0.00000
1002 | 50 | 0.10010
1003 | 52 | 0.20020
1004 | 54 | 0.30030
1005 | 56 | 0.40040
1006 | 58 | 0.50050
1007 | 60 | 0.60060
1008 | 62 | 0.70070
1009 | 64 | 0.80080
1010 | 66 | 0.90090
1011 | 68 | 1.00100
1012 | 70 | 1.10110
1013 | 72 | 1.20120
1014 | 74 | 1.30130
1015 | 76 | 1.40140

Comment (McGiverGim) [reply js/flightlog_fields_presenter.js:514]: Yes, I commented it somewhere to add one decimal to the percentage (more than one is too much).

Another solution is to let the percentage, and add the "real" motor value:

For example: `23 % (46)` or the inverse `46 (23 %)`.

File: js/flightlog.js
Comment (mikeller) [inline js/flightlog.js:1150]: Why are we changing this - we are sending the motor limits explicitly to be able to use the correct values and not having to make assumptions.

File: js/graph_config.js
Comment (mikeller) [inline js/graph_config.js:264]: Again, this should use the actual values for the range.

**Review by mikeller — CHANGES_REQUESTED** [review]

File: js/flightlog_fields_presenter.js
Comment (ctzsnooze) [reply js/flightlog_fields_presenter.js:514]: In true DShot values, we have 1000 steps?  So perhaps one decimal point on the percentage?

File: js/flightlog.js
Comment (ctzsnooze) [reply js/flightlog.js:1150]: Which assumptions are you referring to?

File: js/graph_config.js
Comment (ctzsnooze) [reply js/graph_config.js:264]: What actual values should be used?

**Comment by mikeller** [comment]
Let's establish some facts to facilitate a discussion on this:
- currently, `motor[x]` values in blackbox explorer are rehydrated to be the actual motor output: https://github.com/betaflight/blackbox-log-viewer/blob/7c179dcc8955aa0b0dc208ecab955b9563065007/js/flightlog_parser.js#L1167 - this is all good;
BUT:
- since https://github.com/betaflight/betaflight/pull/10106 `sysConfig.motorOutput[0]` is not the targeted idle speed any more if dynamic idle is enabled, making it unsuitable as a starting point for a 0 - 100% throttle calculation.

So, my proposal to solve this in a generic way is:
- add a `dynamicIdleMotorOffset` header field to the blackbox log, to contain the dynamic idle offset value (or 0 if dynamic idle is not enabled);
- use the existing throttle percentage calculation if dynamic idle is not enabled, or calculate based on the range [`dunamicIdleMotorOffset`...`motorOutput[1]`] if dynamic idle is enabled.

File: js/flightlog.js
Comment (mikeller) [reply js/flightlog.js:1150]: All that are made in this proposal.

File: js/graph_config.js
Comment (mikeller) [reply js/graph_config.js:264]: The ones for the range - or what others can you think of?

Comment (ctzsnooze) [reply js/graph_config.js:264]: I'm not intending to be obtuse to annoy.
I really do not know what range you are referring to.
You could mean the lowest or highest value received in the log for all I know.

File: js/flightlog.js
Comment (ctzsnooze) [reply js/flightlog.js:1150]: That's not actually helping us get anywhere now is it :-)

Comment (ctzsnooze) [reply js/flightlog.js:1150]: I've been using this build for about 200 or more log analyses now.
In day to day use, focusing on idle and motors, it seems perfect to me.
So if you are proposing a change, then please be constructive and indicate what exactly you think should be changed.

**Comment by ctzsnooze** [comment]
This is how, with these changes, a log appears when there is no dynamic idle and a default 5.5% idle, no scaling or anything.

**Comment by ctzsnooze** [comment]
This is an image showing a quad with dynamic idle active, and the default 5.5% idle, showing the motors occasionally going to zero percent throttle.

**Comment by ctzsnooze** [comment]
@mikeller To me this seems exactly how McGyverGim and I, and most people I know, would actually want it displayed.

**Comment by mikeller** [comment]
@ctzsnooze: As I said before, calling something that is based on a range that starts at a point that has got no relation to any physical a 'percentage' is not a good approach. Much rather, the current 'motor output percentage' graph should be fixed so that it displays the output percentage the way all users are used to.

But I can see where you want to go with this, and I think the right approach is to add a new graph for 'motor output value (normalised)' - this graph will be 1000 points both for Dshot and analogue protocols (as this is the range inherent to both protocols), and give you what you want - a scaled down version of the motor percentage based on the adjustments like output scaling and dynamic idle.

**Comment by ctzsnooze** [comment]
I can't comment on the best technical implementation approach.

However from the user perspective, just because something was done a certain way in the past is not a sufficient reason to continue with it in the future.

The original motors graph made sense for the old analog protocols, where there was no absolute zero.  We are not changing anything for those users.  

With digital protocols, and especially with dynamic idle, things are very different.  There is an absolute zero value for the motors, and our idle sits above this at some point.  

The old display method is and was inherently misleading, since it implied that the motor drive was zero when the throttle was zero.  This was never true in any protocol, but in the analog days there was nothing we could do about it.

Now we have digital control protocols and know what the absolute zero is, we can fix this.  It's my view that we need to break from the display of the past so that the user gets 'truthful' information about what the motors are being instructed to do.

I don't agree that we should make an entirely different motors display.  That would be even more confusing.  People would keep asking why one was different from the other.

I am very confident that those who view logs will come to very quickly appreciate the virtues of this change.  It is so much more logical than the old situation, and so obvious, in retrospect, that people will very quickly appreciate the benefit.

So I vote that we should keep the PR as is, other than any technical improvement to the coding, because it does exactly what we should have done long ago.

**Comment by ctzsnooze** [comment]
Mike, I probably look at more log files per day than anyone else on earth.  Honestly.  I deal nearly every day with people who are trying to interpret our logs.  
Unless we have this change, my life gets massively harder.
It is really difficult to be able to show people what happens with dynamic idle, and even what DShot idle actually does, without this change.  
Once we always have the idle shown in the motors graph, the actual idle value going to the motors, lots of things make sense.  It's really hard to explain to people what their idle value does if its invisible.
This gets even more important when they compare motor drive to the rpm values.  If they change DShot idle, with dynamic idle off, their idle rpm changes, but the displayed motor drive at idle does not.  This makes no sense.
Once the idle offset is displayed in the motors tab, a whole lot of things make sense.
As I said before, if the coding here is technically in need of some improvement, I'm all for that, but functionally the outcome is exactly what we need it to be.

**Comment by McGiverGim** [comment]
Is clear that we have different points of view:
- @mikeller wants to show only the minimum "effective" range. To me this is most strange because we have different representations. If the user enables dyn idle, it will have a very different view of the motors than without it. I think a coherent view for all the configurations is better. And you suggest to add more parameters to the header to deal with it. If we have more features in a future that use more effective range than the habitual we need to add more parameters and deal with all of them and maybe have a different limit representation for each one.
- @ctzsnooze and me want to show the motor "possibly" range. The same for all the configurations to maintain coherence. With this we don't need to take into account any feature that reduces or increments the effective range, it will be shown always. For us is good to show in the graphs the limit configured in the motor, the idle (dynamic and static), etc. A blank space in the motors graph is valuable for us if it gives information.

I suppose that the decision to show the current 0-100% effective range in the past was done because nobody, specially in the times that we used analog protocols, thought that in a future we will have features to limit/expand the ranges. Now we have different requisites so we need to adapt to them.

The latest decision is yours @mikeller so if I need to create a new virtual field for this, I can do it, but I don't think is the best solution.

**Comment by mikeller** [comment]
@McGiverGim: My proposal is to add both in parallel:
- keep the 0 - 100% 'motor output percentage' display that is pre-existing, but fix it so that for dynamic idle the percentage can drop below 0% (== idle speed) for users who simply want to see what the motors are doing in their current configuration;
- add a new 'motor output value' that shows the normalised value of 0 - 1000 'motor output units' for users who want to get a view of the 'raw' data output.

**Comment by McGiverGim** [comment]
Yes, that is what I wanted to say with "create a virtual field".

If I'm not wrong, the first part was accomplished with your changes in the firmware part. It changes the header values that are used by the blackbox to decide the range to shown. But I have not tested it.

The second one, is the virtual field to create. But I don't think we must normalize it. Move the dshot/analog values to a normalization of 0-1000 (or 0.0-100.0%) does not give anything here I think. We can't show the real 48-2047 for dshot and the value configured as min-max for analog. I can add both informations if needed (real dshot-analog value and percentage).

But as I said before, I don't think is the best solution. But if the decision has been taken I will look into it later.

**Comment by ctzsnooze** [comment]
@McGiverGim @mikeller 

No, the decision has not been made.  

This is a community project.  

Mike does not have a right to dictate the path of this feature.  No more than anyone else here.  On this topic, less than you and I, to be frank.  

I do not think he intends to override us.  I think he is expressing a personal point of view, one that I respect, but with which I strongly disagree.

If this was a technical coding matter, something to do with code optimisations or other things in which Mike excels, I would defer to his advice.  Absolutely.

This is a usability change for people who use logs every day.  

We have proposed these changes to solve problems with the current archaic way of displaying motors traces.  We both know that it is the right way to go.

I have used this PR now for several weeks and it is invaluable.  I look at maybe 20 logs from different people every day. 

How the motors data is best displayed is something that I know extremely well.  I have tested this with all the variations of throttle limit, throttle boost, motor output limit, dynamic idle, sag compensation and so on.  It was a lot of work.  Some of that effort is in this PR and some in the mixer changes.

I would not have gone to all the trouble to push this unless I felt very strongly that it was worth the effort.

Mike, in all fairness, you have no particular expertise in using logs, or assisting people in interpreting them.  To be blunt, you opinion on this topic is nothing more than your personal opinion, and not one of an expert in the field.  

While I welcome any input, it would be inappropriate to strongly push the archaic method that we need to change, or to have different appearances in different situations that will only add to confusion in future.

This PR is a strong step forward.  If it requires technical improvements, that's fine, but the user level changes we have here should not be messed with.

Please listen to and trust McGiverGim and myself on this.  We have both made our position clear.

If you are concerned that people will be confused by the appearance of the motors panel I will happily handle any such complaints.

**Comment by mikeller** [comment]
Reopening this.
I had a discussion with @ctzsnooze and we agreed that it is an acceptable way forward to do this in a way that this is added as an additional new graph 'Motors (raw)' or suchlike, in addition to the existing graph for which I will be proposing fixes to correctly display dynamic idle.

**Comment by McGiverGim** [comment]
I think I have changed it in the way required, here is the result showing both:

image

I need to do only some minor adjusts:
- Show 1 decimal in the throttle percentage (for both? Only for new?)
- I have added the real raw value next to the percentage in the new. Is that a good decision or must I remove it?

**Comment by McGiverGim** [comment]
Directly uploaded the source with the decimal. Here is the result:
image

**Comment by McGiverGim** [comment]
I will fix the Sonar issues later, so wait until then to review the code.

**Comment by McGiverGim** [comment]
I think that ready to be reviewed. 
I have not fixed a lot of code smells because they are in a code that I have not touched at all and I think it can increase the complexity of the review. If you want that I fix them too is not a problem.

File: js/flightlog.js
Comment (mikeller) [inline js/flightlog.js:1184]: I think it will be better if we extracted the determination if the protocol is Dshot / digital into a function, and used a switch statement like we do in `checkMotorProtocolEnabled()` in `drivers/motor.c` in the firmware.

File: js/flightlog.js
Comment (mikeller) [inline js/flightlog.js:1186]: This is inconsistent with the rest - the range should be 2000 - 1000 to indicate the maximum possible range, just like it is for Dshot.

File: js/flightlog_fields_presenter.js
Comment (mikeller) [inline js/flightlog_fields_presenter.js:534]: And `% absolute` here.

File: js/flightlog_fields_presenter.js
Comment (mikeller) [inline js/flightlog_fields_presenter.js:524]: This should still include the raw value for reference and the percentage should be marked as relative to avoid confusion, so probably `${flightLog.rcMotorRawToPctEffective(value).toFixed(1)} % relative (${value})`.

File: js/flightlog.js
Comment (mikeller) [reply js/flightlog.js:1186]: Actually I was wrong - this should use `maxthrottle` **and `minthrottle`**, and should also use `minthrottle` to normalise the offset.

Comment (McGiverGim) [reply js/flightlog.js:1186]: I was unsure too when I implemented it. The objective of this new motor vision is to show the complete "physical" motor range, from motor stop, to motor at its physical max. In dshot is easy, it has a range from 48 to 2047 for any motor, but in analog I was unsure.

If I remember right, in analog protocols we needed to do a "calibration" that restricts the range that can be pased to the values `minthrottle` to `maxthrottle`, but I was not sure if `minthottle` was the motor stop or was the "analog idle" value.
 
If it is the motor stop, I think you are right and I we need to restrict to it. If it is the "idle" value then I think we need to represent the motor stop too to maintain coherence between both protocols and we need to send lower values too to show the motor stop value, and for this reason I use the 1000 value.

As I say, I was not too sure, it has been so long since I used analog protocols...

File: js/flightlog_fields_presenter.js
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js:534]: I tried to restrict the size to minimum. We have not too much space here, but I will try.

File: js/flightlog_fields_presenter.js
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js:524]: In the old code it has not raw value, but I can add it too without problem. The same concern about the size but I will try.

File: js/flightlog.js
Comment (ctzsnooze) [reply js/flightlog.js:1186]: min_throttle is the lowest sent to any motor while flying, and normally would reflect a stable idle at zero stick.
min command is the value to be sent when the motors should be stopped.
In the old motors display we saw 0% when at minthrottle and 100% when at max throttle, both in graph and numbers.
From the perspective of this PR, it is not quite the same as the DShot situation.  With DShot, zero is always some consistent point below where the motors actually spin.  The relationship between the users idle point and the zero point is reasonably consistent, and has some meaning, especially when dynamic idle is present.
For analog protocols, min_command can be set much lower than the spin-up point.  This is of no particular consequence once flying, and of course dynamic idle cannot be applied.  
By my thinking there is no need, and no benefit, of displaying the offset between min_command and min_throttle in any motors display.  In the past we only ever showed a simple percentage above min_throttle, and we've never had a complaint about that, and I see no virtue in changing this.
For the analog protocols the only useful change could have been to see the actual PWM value that was being sent to the motors, in brackets after the percentage.  Whether it is actually useful to display this, I don't know, it's never been a request and to date I haven't envisaged a use for it.

**Comment by ctzsnooze** [comment]
I very strongly disagree with including a 'legacy' motor page, and adding extra terms such as 'absolute' and 'relative' to deal with the problems that arise from including a 'legacy' motor page.

Having one single updated motors panel as per the original PR will be less confusing than providing two and with extra terminology and extra numbers in brackets.  

The virtue of including include a 'legacy' motors panel is to the user a place they can go to see exactly how it would have looked before.  So if we did include it, then we should not change it in any way.  The proposal seems to be to include it but also change it.  That makes no sense to me.

There is a lot of value in keeping the log display simple and concise.  We should not clutter it with extra values unless they provide a benefit to the user somehow.

If we provide two decimal places for the percentage, that is more than enough accuracy, and for digital protocols will convey the exact value that would have been sent.  Since this PR originally always showed DShot values in 'absolute' form, the percentage to convert to a number comparable to both the idle setpoint and the values that appear in our motors tab.  No other additional number in brackets is required, nor does it add any additional benefit.

I strongly suggest we keep this change clean and simple, making no change to how the analog protocols were displayed, and displaying all digital protocols in absolute range with two decimal points for their percentage.  No extra numbers in brackets, no 'absolute' or 'relative' suffixes.  

I am confident that if we do include a legacy panel, it will get essentially no use and will likely be removed in a year or so.  As such including it, and worry about it, is just a waste of everyones time and energy.

**Comment by ctzsnooze** [comment]
@mikeller could you maybe indicate, clearly, what you think the goal should be here?  I've stated my thoughts, but they disagree with yours, and I think McGyverGim is in the challenging position of having to try to code it.

We have discussed this privately, and I felt that a clear description of your position would be helpful.

My position, as articulated originally in the PR, and modified somewhat following our discussions, is that we should:

- have only one Motors panel, titled, `Motors`, as before.
- display motor drive in percentages to the right of the waveform, as always, but with two decimal points precision
- for analog protocols, the zero point for waveform and percentages would be min_throttle, as always, with the range being from min to max throttle.  No change would be required other than the two decimal points in the percentage number.  
- for digital protocols, the zero point for waveforms and percentages would be a zero DShot drive signal (47 raw) and the 100% point would be full DShot drive (2048 raw).  I don't know if the existing code supports a different DShot range if the user customises it, but ideally it would.  The waveform would show zero when disarmed or motor stop was active, and jump to the defined idle value at idle after arming.  With dynamic idle, motor drive values can go below the DShot idle percentage in flight, but when dynamic idle is not active, they should not.
- no other values would be added in brackets after the percentages, or any other suffixes.

If you want anything different from this, please clearly spell out your vision of how motors data should be displayed, ie if you want to add another panel, how would you name it, how the data would be scaled for analog and digital forms, what ancillary values, if any, besides percentages, would be added, etc.

I would be really grateful if you could give us some clarity on this.

**Comment by McGiverGim** [comment]
Yes please, apart from the technical issues that I need to fix, the final desired result is not clear to me.

I know you have talked about this, so if we can agree the final desired result it will be great.

@mikeller are you ok with the latest message of @ctzsnooze ? Only one motor graph, the one added in this PR, removing the old one?

**Comment by mikeller** [comment]
@McGiverGim: I am ok with the definition that @ctzsnooze gives for the new graph.
Where we disagree on is that I do not think it is the right approach to force users to use the new graph. There are no flash space or performance concerns in the blackbox log viewer, so having the old and new graphs both available comes at a very low cost, and will give users the choice to stay with the graph they are familiar with if they prefer.

So my proposal:
1. keep the existing graph, rename it to 'Motors (legacy)';
2. add in an additional graph with (all from @ctzsnooze):
- display motor drive in percentages to the right of the waveform, as always, but with two decimal points precision
- for analog protocols, the zero point for waveform and percentages would be min_throttle, as always, with the range being from min to max throttle.  No change would be required other than the two decimal points in the percentage number.  
- for digital protocols, the zero point for waveforms and percentages would be a zero DShot drive signal (47 raw) and the 100% point would be full DShot drive (2048 raw).  I don't know if the existing code supports a different DShot range if the user customises it, but ideally it would.  The waveform would show zero when disarmed or motor stop was active, and jump to the defined idle value at idle after arming.  With dynamic idle, motor drive values can go below the DShot idle percentage in flight, but when dynamic idle is not active, they should not.
- no other values would be added in brackets after the percentages, or any other suffixes.

N.b. I will propose a fix to show dynamic idle behaviour in the existing graph once https://github.com/betaflight/betaflight/pull/10307 has been merged.

**Comment by McGiverGim** [comment]
Understood. I will try to do the changes tomorrow.

**Comment by ctzsnooze** [comment]
@mikeller
Thanks for the clarification.  
Which should be the default, going forward - `Motors (legacy)` or `Motors` ?
I would be in complete agreement with you if `Motors` - the new format - was the default.

**Comment by mikeller** [comment]
@ctzsnooze: I think it makes sense to use the new graph as the default for new users.

**Comment by McGiverGim** [comment]
Let's see if I have understood all, this is the final result:

image

File: js/flightlog.js
Comment (McGiverGim) [reply js/flightlog.js]: Added similar method. I didn't like too much the switch but it does not hurt neither...

**Comment by mikeller** [comment]
@McGiverGim: We should leave the existing graph (now 'Motors (legacy)') the way it is, including leaving the motor output values shown in braces after the percentage.

**Comment by McGiverGim** [comment]
The way it is are you saying with the first changes of this PR? In the current master there is no raw value in the motors graph between braces.
But I can change it if you want, something like:
`Motor (Legacy) [1]     20.45 % (454)`

Will be ok?

**Comment by mikeller** [comment]
@McGiverGim: I thought we had them? But yes, having the raw values in the legacy motor graph will be valuable, so please add them.

**Comment by McGiverGim** [comment]
> @McGiverGim: I thought we had them? But yes, having the raw values in the legacy motor graph will be valuable, so please add them.

Added as requested.

**Review by mikeller — APPROVED** [review]

**Pull request merged**
